src/services/users.service.ts

```diff
@@ -36,9 +36,9 @@ export class UsersService {
     return this.userRepository.findOne(id);
   }
 
-  createUser(user: User) {
+  async createUser(user: User) {
     this.hashUserPassword(user);
-    return this.userRepository.insert(user);
+    return await this.userRepository.save(user);
```

Comment:
Please do not use return await.

When you declare an async function, it automatically envelops the return of that function as a promise if it is not a promise already.

Example:

async getName(): Promise<string> {
   return 'ernesto';
}
The example shown above does not return a string. It returns a promise that resolves in a string value.

On the other hand, when you use await inside in an async function you are expecting that function to be executed from the micro-task queue on the Event Loop, and that's results in resolving the promised value.

Example:

```typescript
async printer(): Promise<string> {
   const name: string = await getName(); //  ------->  What you are doing here is expecting the promise to resolve so you have a plain string value and not a Promise that resolves in a string (Promise<string>)
   return name;
}
```
Another way to write the following could be

```typescript
async printer(): Promise<string> {
   return getName();  // ------> Here you are returning the promise of this function so it will be resolved when someone calls the printer function
}
```

So what happens when you use "return await" inside an async function is basically adding additional delay because await pauses the execution until the promise is resolved (or rejected) and then, as you are doing this inside an async function you are forcing to re-promise the value (basically you have a promise, waits it to resolve, and then you are returning a promise again because that's the nature of async functions)